api/pkg/db/migration/202107131802_remove_existing_categories.go

```diff
@@ -0,0 +1,39 @@
+// Copyright © 2021 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package migration
+
+import (
+	"github.com/go-gormigrate/gormigrate/v2"
+	"github.com/tektoncd/hub/api/gen/log"
+	"github.com/tektoncd/hub/api/pkg/db/model"
+	"gorm.io/gorm"
+)
+
+// Removes Existing Categories which are not present
+// in pre-defined Category List
+func removeExistingCategories(log *log.Logger) *gormigrate.Migration {
+
+	return &gormigrate.Migration{
+		ID: "202107131802_remove_existing_categories",
+		Migrate: func(db *gorm.DB) error {
+			if err := db.Where("name IN ?", []string{"Deploy", "Editor", "Language", "Notification", "Others", "Test Framework"}).Delete(&model.Category{}).Error; err != nil {
```

Comment:
could you please delete it permanently

Reply:
Gorm does a soft delete and with soft delete the entry will be just there in the db with the `delete_at` time added for that particular record, so in any of the response we won't get these entries from the db

Reply:
but entries would be remain in the table, Gorm does permanent delete as well